packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -15,9 +17,29 @@ def _strip_undecodable_in_path(path: Path) -> Path:
     return Path(str(path).encode(errors="replace").decode("utf-8"))
 
 
-def _full_file_path_from_dir_and_subdirs(dir_path: Path) -> Iterator[Path]:
+def _get_exclude_regex_rules(exclude_paths: List[Path]) -> List[str]:
+    results: Deque[str] = deque()
+    for path in exclude_paths:
+        if path.is_absolute():
+            results.append(rf"^{path}.*")  # match starts with
+        else:
+            results.append(rf"(.*){path}$")  # match ends with
+    return list(results)
+
+
+def _should_exclude(regex_rules: List[str], path: Path) -> bool:
+    for rule in regex_rules:
+        if re.match(rule, f"{path}"):
+            return False
+    return True
+
+
+def _full_file_path_from_dir_and_subdirs(
+    dir_path: Path, exclude_paths: List[Path] = None
+) -> Iterator[Path]:
+    regex_rules = _get_exclude_regex_rules(exclude_paths if exclude_paths else [])
     for path in dir_path.rglob("*"):
```

Comment:
TIP: notice that you can take advantage of the pattern in ``dir_path.rglob()`` also to exclude! 
the patterns in ``rglob`` are the same as the ones used in https://docs.python.org/3/library/fnmatch.html 
Have a careful read at the possibilities here

Reply:
``_full_file_path_from_dir_and_subdirs``? What is a "full file path"?  is it an [absolute path](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.is_absolute) , a [resolved path](https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve) ??

this function iterates on``Path`` of files under a ``basedir`` and has a list of ignore patterns

```python

def iter_files(basedir: Path, ignore_patterns: Optional[List] = None):
    exclude = exclude or []
    for f in basedir.rglob("*"):
        if f.is_file() and not any(fnmatch(f, p) for p in ignore_patterns):
            yield f

```